packages/store/addon/-private/system/store/record-data-store-wrapper.ts

```diff
@@ -47,59 +91,124 @@ export default class RecordDataStoreWrapper implements IRecordDataStoreWrapper {
     }
   }
 
-  attributesDefinitionFor(modelName) {
-    return this.store._attributesDefinitionFor(modelName);
+  attributesDefinitionFor(modelName: string): AttributesSchema {
+    return this._store._attributesDefinitionFor(modelName);
   }
 
-  relationshipsDefinitionFor(modelName) {
-    return this.store._relationshipsDefinitionFor(modelName);
+  relationshipsDefinitionFor(modelName: string): RelationshipsSchema {
+    return this._store._relationshipsDefinitionFor(modelName);
   }
 
-  inverseForRelationship(modelName, key) {
-    let modelClass = this.store.modelFor(modelName);
-    return this.relationshipsDefinitionFor(modelName)[key]._inverseKey(this.store, modelClass);
+  inverseForRelationship(modelName: string, key: string): string {
+    const modelClass = this._store.modelFor(modelName);
+    const definition = upgradeForInternal(this.relationshipsDefinitionFor(modelName)[key]);
+
+    return definition._inverseKey(this._store, modelClass);
   }
 
-  // TODO Igor David cleanup
-  inverseIsAsyncForRelationship(modelName, key) {
-    let modelClass = this.store.modelFor(modelName);
-    return this.relationshipsDefinitionFor(modelName)[key]._inverseIsAsync(this.store, modelClass);
+  inverseIsAsyncForRelationship(modelName: string, key: string): boolean {
+    const modelClass = this._store.modelFor(modelName);
+    const definition = upgradeForInternal(this.relationshipsDefinitionFor(modelName)[key]);
+
+    return definition._inverseIsAsync(this._store, modelClass);
   }
 
-  notifyPropertyChange(modelName, id, clientId, key) {
-    let internalModel = this.store._getInternalModelForId(modelName, id, clientId);
-    internalModel.notifyPropertyChange(key);
+  notifyPropertyChange(modelName: string, id: string | null, clientId: string, key: string): void;
```

Comment:
Why do we need multiple signatures here. Fwiw, the new prettier line length thing, is making this much noisier for me to read

Reply:
I've fixed up prettier to ensure longer line lengths